File: rhine/src/main/java/com/qingmei2/rhine/http/cache/CacheProviders.java
package com.qingmei2.rhine.http.cache;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.rx_cache2.internal.RxCache;
import lombok.Getter;

@Singleton
public class CacheProviders {

    @Getter
    private UserInfoCacheProviders userInfoCacheProviders;

    @Inject
    public CacheProviders(RxCache rxCache) {
        this.userInfoCacheProviders = rxCache.using(UserInfoCacheProviders.class);
    }

}

File: rhine/src/main/java/com/qingmei2/rhine/http/cache/CacheProviders.kt
package com.qingmei2.rhine.http.cache

import javax.inject.Inject
import javax.inject.Singleton

import io.rx_cache2.internal.RxCache
import lombok.Getter

@Singleton
class CacheProviders @Inject
constructor(rxCache: RxCache) {

    @Getter
    private val userInfoCacheProviders: UserInfoCacheProviders

    init {
        this.userInfoCacheProviders = rxCache.using(UserInfoCacheProviders::class.java)
    }

}
